frontend/src/pages/ProgressionChecker/GridView/GridView.jsx

```diff
@@ -75,14 +81,92 @@ const GridView = ({ isLoading, structure }) => {
   };
 
   useEffect(() => {
-    setPastCourses(getPastCourses(years, startYear, courses));
+    setPlannerCourses(getFormattedPlannerCourses(years, startYear, courses));
     generateGridStructure();
   }, [isLoading, structure, years, startYear, courses]);
 
   const handleCourseLink = (courseCode) => {
     navigate("/course-selector");
     dispatch(addTab(courseCode));
   };
+
+  const courseBadge = (course) => {
+    if (course.unplanned) {
+      return (
+        <Badge
+          key={course.key}
+          count={(
+            <Tooltip title="Course added but not planned">
+              <div className="courseBadgeIcon">!</div>
+            </Tooltip>
+          )}
+        >
+          <Button
+            className="checkerButton"
+            type="primary"
+            style={{ background: "#FFF", color: "#9254de" }}
+            key={course.key}
+            onClick={() => handleCourseLink(course.key)}
+          >
+            {course.key}: {course.title}
+          </Button>
+        </Badge>
+      );
+    }
+
+    if (course.past) {
+      return (
+        <Button
+          className="checkerButton"
+          type="primary"
+          key={course.key}
+          onClick={() => handleCourseLink(course.key)}
+        >
+          {course.key}: {course.title}
+        </Button>
+      );
+    }
+
+    // for future courses planned
+    // course.past can be undefined if not in term planner thus check for false
+    if (course.past === false) {
```

Comment:
doesn't seem like this works because by removing planned I'm using ```!unplanned``` and since it evaluates to true even when the course isn't planned, the incorrect courses are getting classified as future planned courses

Reply:
Ohhh okie get what you mean then